docs/cvl/cvl2/changes.md

```diff
@@ -0,0 +1,583 @@
+longer useful.  If you find that you do need one of these features, contact
+Certora support.
+
+### Methods entries for sighashes
+
+In CVL 1, you could write a sighash instead of a method identifier in the
+`methods` block.  This feature is no longer supported.  You will need to have
+the name and argument types of the called method in order to provide an entry.
+
+```{todo}
+If you do not change this, you will see the following error:
+```
+
+### `invoke`, `sinvoke`, and `call`
+
+Older versions of CVL had special syntax for calling contract functions:
+ - `invoke f(args);` should be replaced with `f(args);`.
+ - `sinvoke f(args);` should be replaced with `f@withrevert(args);`
+ - `call f(args)` should be replaced with `f(args)`.
+
+```{todo}
+If you do not change this, you will see the following error:
+```
+
+### `static_assert` and `static_require`
+
+These deprecated aliases for `assert` and `require` are being removed; replace
+them with `assert` and `require`.
+
+```{todo}
+If you do not change this, you will see the following error:
+```
+
+### `invoke_fallback`
+
+The `invoke_fallback` syntax is no longer supported; there is no longer a way
+to directly invoke the fallback method.
+
+```{todo}
+If you do not change this, you will see the following error:
+```
+
+
+### Havocing `calldataarg` variables
+
+In CVL 1, you could write the following:
+
+```cvl
+calldataarg args; env e;
+f(e, args);
+
+havoc args;
+g(e, args);
+```
+
+You can no longer write `havoc x` where `x` is any variable of type `calldataarg`.
+
+Instead, replace the havoced variable with a new variable.
+
+```{todo}
+If you do not change this, you will see the following error:
+```
+
+### Destructuring syntax for struct returns
+
+In CVL 1, if a contract function returned a struct, you could use a
+destructuring syntax to get the return value in your spec.  For example,
+consider the following contract:
+
+```solidity
+contract Example {
+    struct S {
+        uint firstField;
+        uint secondField;
+        bool thirdField;
+    }
+
+    function f() returns(S) { ... }
+}
+```
+
+To access the return value of `f` in CVL 1, you could write the following:
+
+```cvl
+uint x; uint y; bool z;
+x, y, z = f();
+```
+
+This syntax is no longer supported.  Instead, you should declare a variable with
+the struct type:
+
+```cvl
+Example.S result = f();
+uint x = result.firstField;
+```
+
+```{todo}
+If you do not change this, you will see the following error:
+```
+
+### `bytes[]` and `string[]`
+
+In CVL 1, you could declare variables of type `string[]` and `bytes[]`.  You can
+no longer use these types in CVL.
+
+You can still declare contract methods that use these types in the `methods`
+block.  However, you can only call methods that take one of these types as an
+argument by passing a `calldataarg` variable, and you cannot access the return
+value of a method that returns one of these types.
+
+```{todo}
+Determine whether you can call with `_`.
```

Comment:
Very likely not. I believe the mechanism is to infer the type and generate a variable of the type internally.